### PR TITLE
Make NodeController recognize deletion tombstones

### PR DESCRIPTION
**Pull request by davidopp**
cc/ @lavalamp @gmarek

``` release-note
Make NodeController recognize deletion tombstones
```

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/34786)

### CODE REVIEW

File: pkg/controller/node/nodecontroller.go [hunk 1]
```diff
@@ -236,6 +236,7 @@ func NewNodeController(
 	podInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
 		AddFunc:    nc.maybeDeleteTerminatingPod,
 		UpdateFunc: func(_, obj interface{}) { nc.maybeDeleteTerminatingPod(obj) },
+		DeleteFunc: nc.maybeDeleteTerminatingPod,
```
Comment (wojtek-t) [inline pkg/controller/node/nodecontroller.go]: I don't understand this line. If we observed deletion of pod, why we would even want to try delete it again?

Comment (davidopp) [inline pkg/controller/node/nodecontroller.go]: Oh sorry, I had added a pending comment that I wasn't sure this line was right, but I forgot to hit "review changes." (After un-learning the need to "send changes" now I am having trouble adapting to needing to do it...). I'll delete it.

**Comment by davidopp** [comment]
OK, PTAL.

**Comment by wojtek-t** [comment]
Can you please add release-note? (I think we need it for things to be cherrypicked).

LGTM

**Comment by davidopp** [comment]
I've set `release-note` label just now, which will use the issue title as the release note, IIUC?

**Comment by gmarek** [comment]
Actually I'm not a fan of this PR - I'd rather do it directly when handler is registered.

**Pull request merged**